vscode-react/src/components/aKDeploymentTableBody.component.tsx

```diff
@@ -116,6 +116,16 @@ export const AKDeploymentTableBody = ({ deployments }: { deployments?: Deploymen
 		hidePopper();
 	};
 
+	const deleteDeploymentDisplayPopper = (event: React.MouseEvent<HTMLDivElement>, deployment: Deployment) => {
+		if (deployment.state === DeploymentState.ACTIVE_DEPLOYMENT) {
+			return;
+		}
+		const refElement = event.currentTarget;
+		showPopper("deploymentDelete");
+		deletePopperElementRef.current = refElement;
+		setDeleteDeploymentId(deployment.deploymentId);
```

Comment:
what setDeleteDeploymentId do ?

Reply:
The flow goes like this:
1. The user clicks the delete icon, which opens the deletePopper.
2. When the popper is opened, the `deploymentId` is being saved in the local state for further action.
3. When the user approves that he wants to delete the deployment, the request for the extension is being sent with the `deploymentId` saved when the modal has been opened: https://github.com/autokitteh/vscode-extension/blob/bc315de55bc59708a99a7daccd0bf30a2bdd407e/vscode-react/src/components/aKDeploymentTableBody.component.tsx#L109C1-L110C1

So we are saving the deploymentId to send it when the user approves the deletion.